go/libraries/doltcore/merge/merge_prolly_rows.go

```diff
@@ -89,7 +89,13 @@ func mergeProllyTable(ctx context.Context, tm *TableMerger, mergedSch schema.Sch
 
 		// After we migrate the data on the left-side to the new, merged schema, we reset
 		// the left mapping to an identity mapping, since it's a direct mapping now.
-		valueMerger.leftMapping = val.NewIdentityOrdinalMapping(len(valueMerger.leftMapping))
+		// However, columns that did not exist on the left schema shouldn't be updated, because they
+		// still don't exist on the migrated data.
```

Comment:
> I'm mostly just wondering if the logic will be simpler later if we could always assume left had all the new columns after we migrated the data.

For most cases it doesn't matter, because of what you point out:  when nullable columns are added to the end of the schema, we don't rewrite the table. The logic for using mappings to read from tuples is doesn't care if the mapping points to an index past the end of the tuple or if the mapping contains `-1` (indicating that the column doesn't exist in the tuple.) In both cases, attempting to retrieve the value from the tuple returns nil/Null.

However, there's one case where the distinction matters: when attempting to automatically resolve data conflicts in `processColumn`, we need to be able to differentiate between the case where the left schema has the column (and it's NULL) vs the case where the left schema doesn't have the column. We can't inspect the tuple because null values at the end of the tuple are omitted. And we can't use the schema anymore because by migrating the tuple here it no longer matches the schema. So we need to distinguish these cases in the mapping in order to correctly do data merges.